Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixInfo.tsx

```diff
@@ -0,0 +1,121 @@
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader, Table } from "reactstrap";
+
+interface IProps {
+    bigfixId: string;
+}
+
+interface IState {
+    bigfixModal: boolean;
+    computerInfo: object;
+    isFetched: boolean;
+}
+
+export default class EquipmentDetails extends React.Component<IProps, IState> {
+    public static contextTypes = {
+        fetch: PropTypes.func,
+        permissions: PropTypes.array,
+        router: PropTypes.object,
+        team: PropTypes.object
+    };
+
+    public state = { bigfixModal: false, computerInfo: {}, isFetched: false };
```

Comment:
why is this `public state = {...}` and not `this.state = {...}`?

Reply:
we can declare state outside of the constructor as well. but I am not sure if we should have it as public, private or protected. I will change it to private since it only belongs to the class.

Reply:
yes, it is good convention to initialize state in the constructor. fixed